test/serialization.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly/core';
+
+const storageKey = 'mainWorkspace';
+
+/**
+ * Saves the state of the workspace to browser's local storage.
+ * @param {Blockly.Workspace} workspace Blockly workspace to save.
+ */
+export const save = function(workspace) {
+  const data = Blockly.serialization.workspaces.save(workspace);
+  window.localStorage?.setItem(storageKey, JSON.stringify(data));
+};
+
+const defaultData = {
+  'blocks': {
+    'languageVersion': 0,
+    'blocks': [
+      {
+        'type': 'p5_setup',
```

Comment:
Is it actually helpful to have these things by default? Or does that mess with testing? I think it would be better to have a button that loads some state rather than saying you must always have this state. This makes it kind of inflexible for researchers.

Reply:
I think it should start with *some* state by default, because it reduces the work needed for a test user to start using Blockly--no one has to come through and set up the blocks for them.

I am interested in having buttons that let you load a few different scenarios, with instructions on how to interact with those scenarios (adding blocks vs editing blocks, etc). I thought that could be a separate feature.